pulp_rpm/app/modulemd.py

```diff
@@ -67,14 +67,9 @@ def split_modulemd_file(file):
         decompressed_path = os.path.join(tf, "modulemd.yaml")
         cr.decompress_file(file.path, decompressed_path, cr.AUTO_DETECT_COMPRESSION)
         with open(decompressed_path) as modulemd_file:
-            module = ""
-            for line in modulemd_file:
-                if line.startswith("---"):
-                    if module:  # avoid first empty yield in the beginning of document
-                        yield module.strip("\n")
-                        module = ""
-                module += line
-            yield module.strip("\n")
+            documents = modulemd_file.read().split("---")
```

Comment:
Discussed w/ dralley the memory-implications here (ie, reading the entire modulemd file into memory at once and returning it as a list, rather than using a generator). modulemd files are generally of "manageable" size, and the code-simplification feels like a reasonable tradeoff.